packages/beacon-node/src/api/impl/beacon/blocks/index.ts

```diff
@@ -37,6 +39,127 @@ export function getBeaconBlockApi({
   network,
   db,
 }: Pick<ApiModules, "chain" | "config" | "metrics" | "network" | "db">): ServerApi<routes.beacon.block.Api> {
+  const publishBlock: ServerApi<routes.beacon.block.Api>["publishBlock"] = async (
+    signedBlockOrContents,
+    opts: PublishBlockOpts = {}
+  ) => {
+    const seenTimestampSec = Date.now() / 1000;
+    let blockForImport: BlockInput, signedBlock: allForks.SignedBeaconBlock, signedBlobs: deneb.SignedBlobSidecars;
+
+    if (isSignedBlockContents(signedBlockOrContents)) {
+      ({signedBlock, signedBlobSidecars: signedBlobs} = signedBlockOrContents);
+      blockForImport = getBlockInput.postDeneb(
+        config,
+        signedBlock,
+        BlockSource.api,
+        // The blobsSidecar will be replaced in the followup PRs with just blobs
+        blobSidecarsToBlobsSidecar(
+          config,
+          signedBlock,
+          signedBlobs.map((sblob) => sblob.message)
+        ),
+        null
+      );
+    } else {
+      signedBlock = signedBlockOrContents;
+      signedBlobs = [];
+      // TODO: Once API supports submitting data as SSZ, replace null with blockBytes
+      blockForImport = getBlockInput.preDeneb(config, signedBlock, BlockSource.api, null);
+    }
+
+    // check what validations have been requested before broadcasting and publishing the block
+    // TODO: add validation time to metrics
+    const broadcastValidation = opts.broadcastValidation ?? routes.beacon.BroadcastValidation.none;
+    // if block is locally produced, full or blinded, it already is 'consensus' validated as it went through
+    // state transition to produce the stateRoot
+    const slot = signedBlock.message.slot;
+    const blockRoot = toHex(chain.config.getForkTypes(slot).BeaconBlock.hashTreeRoot(signedBlock.message));
+    const blockLocallyProduced =
+      chain.producedBlockRoot.has(blockRoot) || chain.producedBlindedBlockRoot.has(blockRoot);
+    const valLogMeta = {broadcastValidation, blockRoot, blockLocallyProduced, slot};
+
+    switch (broadcastValidation) {
+      case routes.beacon.BroadcastValidation.none: {
+        if (blockLocallyProduced) {
+          chain.logger.debug("No broadcast validation requested for the block", valLogMeta);
+        } else {
+          chain.logger.warn("No broadcast validation requested for the block", valLogMeta);
+        }
+        break;
+      }
+      case routes.beacon.BroadcastValidation.consensus: {
```

Comment:
I think this PR should implement the different broadcast validations

Reply:
would prefer to do followups unless you feel strongly about this